qcodes/tests/parameter/test_parameter_scale_offset.py

```diff
@@ -172,3 +172,20 @@ def test_scale_and_offset_raw_value_iterable_for_set_cache(values,
         event('Scale is array and also offset')
     if isinstance(scales, Iterable) and not isinstance(offsets, Iterable):
         event('Scale is array but not offset')
+
+
+def test_data_return_type_after_setting_scale_and_offset():
```

Comment:
perhaps a more specific/explicit test name?
```suggestion
def test_numpy_array_valued_parameter_preserves_type_if_scale_and_offset_are_set():
```